packages/react-accordion/src/components/AccordionHeader/useAccordionHeader.ts

```diff
@@ -57,12 +59,14 @@ export const useAccordionHeader = (props: AccordionHeaderProps, ref: React.Ref<H
     }),
     button: {
       ...buttonShorthand,
-      onClick: useEventCallback(ev => {
-        buttonShorthand.onClick?.(ev);
-        if (!ev.defaultPrevented) {
-          onAccordionHeaderClick(ev);
-        }
-      }),
+      onClick: useEventCallback(
+        (ev: React.MouseEvent<HTMLButtonElement & HTMLDivElement & HTMLSpanElement & HTMLAnchorElement>) => {
```

Comment:
Does it make sense to use `HTMLElement` here, unless there's a big difference between the event props between different types of elements

Reply:
Unfortunately because `ElementShorthandProps` has strongly-typed element types for all of the events, this has to be a explicit  `&` of all of the supported types. `buttonShorthand.onClick?.(ev)` fails to compile without it, because the event type passed in can only be one of the supported types; not any HTMLElement.